### ISSUE TITLE
Generate debug info, potentially mapping from source language debuginfo to machine code

### ISSUE THREAD

**Issue by nagisa** [near/wasmer#110]
It would be interesting and potentially beneficial to the end users' debugging stories if we generated debuginfo for the machine code we produce.

This is a potentially interesting problem to research because the debug info generated would need to map not from machine code back to input wasm, like in the usual compilation scenarios, but rather from the machine code back to the the code from which the wasm code was produced.

**Comment by matklad** [comment]
>The ultimate goal should be that running gdb --args standalone-vm-runner contract.wasm would allow to see source listings of the source language, stepping through source language statements, etc.

Yeah, we need something like this. This shouldn't be `standalone-vm-runner` though, but rather something like `neard-sandbox --call-contract contract.wasm` or something along those lines. We should centralize on sandbox as one-stop-shop binary for contract test&debugging. `standalone-vm-runner` isn't really a public and supported interface at the moment.

More generally, we should work on making some sort-of public interface here. It'd be cool if, eg, one could run contract tests via 

```console
$ CARGO_RUNNER="neard-sandbox --test-contract" cargo test --target wasm32-unknown-unknown
```

Feels like a big undertaking, no one is actively working on it right now.
